src/components/PostPicker/index.js

```diff
@@ -0,0 +1,389 @@
+		return select( 'core' ).getEntityRecords( 'postType', postType, queryArgs ) ?? [];
+	}, [ postType, queryArgs ] );
+
+	const isResolving = useSelect( ( select ) => {
+		return select( 'core/data' ).isResolving( 'core', 'getEntityRecords', [ 'postType', postType, queryArgs ] );
+	} );
+
+	const reorderUp = ( postId ) => {
+		const index = values.indexOf( postId );
+
+		// Item not found or already at the first position
+		if ( index === -1 || index === 0 ) {
+			return;
+		}
+
+		onChange( [
+			...values.slice( 0, index - 1 ),
+			values[ index ],
+			values[ index - 1 ],
+			...values.slice( index + 1 ),
+		] );
+	};
+
+	const reorderDown = ( postId ) => {
+		const index = values.indexOf( postId );
+
+		// Item not found or already at the last position
+		if ( index === -1 || index === values.length - 1 ) {
+			return;
+		}
+
+		onChange( [
+			...values.slice( 0, index ),
+			values[ index + 1 ],
+			values[ index ],
+			...values.slice( index + 2 ),
+		] );
+	};
+
+	return (
+		<div style={ {
+			// Overflow at top.
+			marginTop: -24,
+			paddingTop: 24,
+			// Offset to display checkbox focus outline.
+			paddingLeft: 4,
+			marginLeft: -4,
+		} }>
+			{
+				( isResolving && <Spinner /> )
+					|| ( queriedPosts.length < 1 &&
+						<Notice
+							isDismissible={ false }
+						>
+							{ __( 'No results found' ) }
+						</Notice>
+					)
+					|| (
+						queriedPosts.map( post => (
+							<div
+								style={ {
+									display: 'grid',
+									gridTemplateColumns: '1fr auto',
+									marginRight: -2,
+									paddingRight: 2,
+								} }
+							>
+								<CheckboxControl
+									key={ post.id }
+									checked={ values.includes( post.id ) }
+									label={ post.title?.rendered || __( '(No title)', 'block-editor-components' ) }
+									onChange={ ( checked ) => {
+										if ( checked ) {
+											onChange( [ ...values, post.id ] );
+										} else {
+											onChange( values.filter( value => ( value !== post.id ) ) );
+										}
+									} }
+								/>
+								{ isSortable && (
+									<ButtonGroup>
+										<Button
+											icon={ 'arrow-up-alt2' }
+											iconSize={ 12 }
+											isSmall
+											label={ __( 'Move up', 'block-editor-components' ) }
+											variant="secondary"
+											onClick={ () => reorderUp( post.id ) }
+										/>
+										<Button
+											icon={ 'arrow-down-alt2' }
+											iconSize={ 12 }
+											isSmall
+											label={ __( 'Move down', 'block-editor-components' ) }
+											variant="secondary"
+											onClick={ () => reorderDown( post.id ) }
+										/>
+									</ButtonGroup>
+								) }
+							</div>
+						) )
+					)
+			}
+		</div>
+	);
+}
+
+/**
+ * Post Picker Browse Post Panel.
+ *
+ * @param props
+ * @returns {ReactNode} Component
+ */
+function BrowsePanel( props ) {
+	const {
+		postType,
+		onChange,
+		values,
+		taxonomies,
+	} = props;
+
+	const [ search, setSearch ] = useState();
```

Comment:
does this need an initial value?

Reply:
Hmm... maybe. 

If `search` is an empty string, or `null`, and that is passed as part of the `queryArgs` to `getEntityRecords` then the API request includes `?search=`. I don't _think_ that's a problem... I assume the Rest API is ignoring an empty search query. However it does feel _neater_ if we dont' send this arg when it's empty. 

But -  you can also get issues if values are undefined. Passing them straight to an `input` as the value, will cause a warning like `Warning: A component is changing an uncontrolled input of type text to be controlled.`. 

So probably it is better to be consistent about the type of data we're dealing with, setting an empty string as a default, and then handle that when defining query args.